packages/matter.js/src/cluster/GeneralDiagnosticCluster.ts

```diff
@@ -0,0 +1,209 @@
+    /** human-readable (displayable) name for the network interface */
+    name: TlvField(0, TlvString.bound({ maxLength: 32 })),
+
+    /** indicate if the Node is currently advertising itself operationally on this network interface */
+    isOperational: TlvField(1, TlvBoolean),
+
+    /** indicate whether the Node is currently able to reach off-premise services it uses by utilizing IPv4 */
+    offPremiseServicesReachableIPv4: TlvField(2, TlvNullable(TlvBoolean)), /* default null */
+
+    /** indicate whether the Node is currently able to reach off-premise services it uses by utilizing IPv4 */
+    offPremiseServicesReachableIPv6: TlvField(3, TlvNullable(TlvBoolean)),  /* default null */
+
+    /** SHALL contain the current link-layer address for a 802.3 or IEEE 802.11-2020 network interface and 
+     * contain the current extended MAC address for a 802.15.4 interface 
+    */
+    hardwareAddress: TlvField(4, TlvString.bound({ maxLength: 8 })),
+
+    /** list of the IPv4 addresses that are currently assigned to the network interface */
+    iPv4Addresses: TlvField(5, TlvArray(TlvString.bound({ maxLength: 4 }))),
+
+    /** list of the unicast IPv6 addresses that are currently assigned to the network interface. */
+    iPv6Addresse: TlvField(5, TlvArray(TlvString.bound({ maxLength: 8 }))),
+
+    /** indicate the type of the interface */
+    type: TlvField(6, TlvEnum<InterfaceType>()),
+});
+
+/** SHALL indicate a change in the set of hardware faults currently detected by the Node
+   * @see {@link MatterCoreSpecificationV1_0} §11.11.9.1
+ */
+const HardwareFaultChangeEvent = {
+    current: TlvField(0, TlvArray(TlvEnum<HardwareFault>(), { maxLength: 11 })),
+    previous: TlvField(1, TlvArray(TlvEnum<HardwareFault>(), { maxLength: 11 })),
+};
+
+/** SHALL indicate a change in the set of radio faults currently detectedby the Node
+  * @see {@link MatterCoreSpecificationV1_0} § 11.11.9.2
+ */
+const RadioFaultChangeEvent = {
+    current: TlvField(0, TlvArray(TlvEnum<HardwareFault>(), { maxLength: 7 })),
+    previous: TlvField(1, TlvArray(TlvEnum<HardwareFault>(), { maxLength: 7 })),
+};
+
+/** SHALL indicate a change in the set of network faults currently detected by the Node
+  * @see {@link MatterCoreSpecificationV1_0} § 11.11.9.3
+*/
+const NetworkFaultChangeEvent = {
+    current: TlvField(0, TlvArray(TlvEnum<HardwareFault>(), { maxLength: 4 })),
+    previous: TlvField(1, TlvArray(TlvEnum<HardwareFault>(), { maxLength: 4 })),
+};
+
+/** SHALL indicate the reason that caused the device to start-up
+  * @see {@link MatterCoreSpecificationV1_0} § 11.11.7.4  
+*/
+const BootReasonEvent = {
+    bootReason: TlvField(0, TlvEnum<BootReason>()),
+};
+
+/** Provide a means for certification tests to trigger some test-plan-specific events 
+  * @see {@link MatterCoreSpecificationV1_0} § 11.11.8.1 
+*/
+const TlvTestEventTriggerRequest = TlvObject({
+    enableKey: TlvField(0, TlvString.bound({ length: 16 })),
```

Comment:
maybe we also have a StringMax16? :-)

Reply:
We don't appear to have 16, only 32, 64 and 256